From 74ce954c44b03dea8bb99f25b1874dfa92d887d0 Mon Sep 17 00:00:00 2001
From: jjacky <i.am.jack.mail@gmail.com>
Date: Sat, 6 Apr 2013 17:13:53 +0200
Subject: [PATCH] Unreferencing a GtkPlug could cause abort; Fix #697427

Since 3d4cd4db and the use of gtk_widget_(un)register_window in GtkPlug a bug
could manifest, because the user_data associated with a GdkWindow is checked
beforehand.

But after a reparent, when a new socket_window is used, it wasn't registered,
hence the failed assert.

Signed-off-by: jjacky <i.am.jack.mail@gmail.com>
---
 gtk/gtkplug.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gtk/gtkplug.c b/gtk/gtkplug.c
index 01fc979..61955c0 100644
--- a/gtk/gtkplug.c
+++ b/gtk/gtkplug.c
@@ -931,6 +931,7 @@ gtk_plug_filter_func (GdkXEvent *gdk_xevent,
 		if (!priv->socket_window) /* Already gone */
 		  break; /* FIXME: shouldn't this unref the plug? i.e. "goto done;" instead */
 	      }
+            gtk_widget_register_window (GTK_WIDGET (plug), priv->socket_window);
 
             if (priv->grabbed_keys)
               {
-- 
1.8.2

